File: FitBook.Model/Responses/TrainingEquipment/TrainingEquipmentResponse.cs
namespace FitBook.Model.Responses.TrainingEquipment;

public class TrainingEquipmentResponse : IEntityResponse
{
    public int Id { get; set; }
    public bool IsRequired { get; set; }
    public string? Note { get; set; }
    public int TrainingId { get; set; }
    public int EquipmentId { get; set; }
    public string EquipmentName { get; set; } = string.Empty;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: UI/fitbook_desktop/lib/models/responses/training_equipment_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'training_equipment_response.g.dart';

@JsonSerializable()
class TrainingEquipmentResponse {
  TrainingEquipmentResponse({
    required this.id,
    required this.isRequired,
    this.note,
    required this.trainingId,
    required this.equipmentId,
    required this.equipmentName,
    required this.createdAtUtc,
    this.updatedAtUtc,
  });

  final int id;
  final bool isRequired;
  final String? note;
  final int trainingId;
  final int equipmentId;
  final String equipmentName;
  final DateTime createdAtUtc;
  final DateTime? updatedAtUtc;

  factory TrainingEquipmentResponse.fromJson(Map<String, dynamic> json) =>
      _$TrainingEquipmentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$TrainingEquipmentResponseToJson(this);
}
